keystrokes/src/components/Character/CharacterDetails/ManualKeybindInputs.js

```diff
@@ -0,0 +1,368 @@
+import React from 'react';
+import { red } from '@material-ui/core/colors';
+import {
+	Tooltip,
+	Button,
+	Grid,
+	TextField,
+	Typography,
+	MenuItem,
+	Avatar,
+} from '@material-ui/core';
+import WarningIcon from '@material-ui/icons/Warning';
+import PropTypes from 'prop-types';
+import styleGuide from '../../../stylesheets/style';
+import { verifyKey, validatePress } from '../../utils/utils';
+import { targetting, mods } from '../../../config/constants';
+
+ManualKeybindInputs.propTypes = {
+	spec: PropTypes.string.isRequired,
+	invalidBind: PropTypes.array.isRequired,
+	setInvalidBind: PropTypes.func.isRequired,
+	onSubmit: PropTypes.func.isRequired,
+	allKeybinds: PropTypes.array.isRequired,
+	keybinding: PropTypes.object.isRequired,
+	setKeybinding: PropTypes.func.isRequired,
+	checkIfInvalidAndAdd: PropTypes.func.isRequired,
+	Spells: PropTypes.object.isRequired,
+};
+
+function ManualKeybindInputs({
+	spec,
+	invalidBind,
+	setInvalidBind,
+	onSubmit,
+	allKeybinds,
+	keybinding,
+	setKeybinding,
+	checkIfInvalidAndAdd,
+	Spells,
+}) {
+	const classes = styleGuide();
+
+	function handleKeyPress(e) {
+		if (!e.metaKey) {
+			e.preventDefault();
+		}
+		const newKey = e.code.toLowerCase().replace(/digit|key/i, '');
+		if (validatePress(newKey)) {
+			setKeybinding({
+				...keybinding,
+				Key: verifyKey(newKey),
+			});
+		} else if (newKey === 'backspace' || newKey === 'delete') {
+			setKeybinding({
+				...keybinding,
+				Key: '',
+			});
+		}
+		checkIfInvalidAndAdd({
+			...keybinding,
+			Key: verifyKey(newKey),
+		});
+	}
+
+	return (
+		<Grid
+			container
+			className={classes.paddingTop}
+			justify="center"
+			alignItems="center"
+		>
+			<Grid item className={classes.keybindingOptions}>
+				<TextField
+					className={classes.button}
+					select
+					id={'spell-selector'}
+					variant="outlined"
+					value={keybinding.Spell || ''}
+					label="Spell"
+					onChange={event => {
+						setKeybinding({
+							Spell: event.target.value,
+							Target: null,
+							Mod: null,
+							Key: null,
+						});
+						setInvalidBind([]);
+					}}
+				>
+					{Object.entries(Spells)
+						.filter(spell => spell[1].spec.includes(spec))
+						.map(spell => {
+							const existingSpellBinds = allKeybinds.filter(
```

Comment:
Hmmm I'd consider using some instead here and having a bool
Then down below where you have the map you can do the filter there since it's the only real place it's used

Reply:
its used in a few different places in this scope, down with avatar and the value in avatar, i think i do it this way cause i put a number down inside the avatar itself